lens-cube/src/main/java/org/apache/lens/cube/parse/CandidateUtil.java

```diff
@@ -43,17 +43,26 @@ public static boolean isMeasureExpressionAnswerable(ASTNode exprNode, Candidate
   /**
    * Returns true if the Candidate is valid for the given time range based on its start and end times.
    * @param candidate
-   * @param timeRange
+   * @param timeRanges
    * @return
    */
-  public static boolean isValidForTimeRange(Candidate candidate, TimeRange timeRange) {
-    return (!timeRange.getFromDate().before(candidate.getStartTime()))
-      && (!timeRange.getToDate().after(candidate.getEndTime()));
+  public static boolean isValidForTimeRange(Candidate candidate, List<TimeRange> timeRanges) {
+    for (TimeRange timeRange : timeRanges) {
+      if (!(!timeRange.getFromDate().before(candidate.getStartTime()))
```

Comment:
if (timeRange.getFromDate().before(candidate.getStartTime() ||
          timeRange.getToDate().after(candidate.getEndTime()) {
return false;
}

Will this work ?